Update copyrights on all files

pavanbalaji opened the pull request:
## Pull Request Description

Update copyrights on all files

## Expected Impact

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

pavanbalaji commented:
test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

pavanbalaji commented:
This PR only has the emacs mode lines for all files.  If someone can tell me the vim mode lines, I can add them too.

pavanbalaji commented:
test:whitespace

pavanbalaji commented:
test:jenkins/warnings

pavanbalaji commented:
test:jenkins/warnings

pavanbalaji commented:
test:jenkins/warnings

pavanbalaji commented:
test:jenkins/ch3/most

pavanbalaji commented:
test:jenkins/ch3/most

pavanbalaji commented:
test:jenkins/ch3/tcp

pavanbalaji commented:
test:jenkins/ch3/tcp

pavanbalaji commented:
test:jenkins/ch3/tcp

pavanbalaji commented:
test:jenkins/ch3/tcp

pavanbalaji commented:
test:jenkins/ch3/tcp

pavanbalaji commented:
@hzhou or @raffenet 

Can you take a look at what's going on here?  Jenkins seems to be misbehaving.

hzhou commented:
> Can you take a look at what's going on here? Jenkins seems to be misbehaving.

Something changed in github:
```
 > git fetch --tags --progress http://github.com/pmodels/mpich +refs/pull/*:refs/remotes/origin/pr/* # timeout=10
ERROR: Error fetching remote repo 'origin'
```
, which used to work, but stopped working. I have a workaround -- configure the tests to fetch from the author branch directly instead. I'll update the review tests.

hzhou commented:
test:jenkins/ch3/tcp

hzhou commented:
I think the review tests are now working.

pavanbalaji commented:
> I think the review tests are now working.

It still seems to be failing, but with a different error now.

pavanbalaji commented:
> > I think the review tests are now working.
> 
> It still seems to be failing, but with a different error now.

Nevermind.  It seems to be working.  I was looking at the old tests.

pavanbalaji commented:
It seems to be working fine now.  Running the remaining tests:

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

pavanbalaji commented:
Forgot to force push my actual changes.  Trying again.

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

hzhou commented:
>  test/mpi/maint/jenkins/set-xfail.sh → test/mpi/maint/jenkins/set-xfail.zsh 

Does this script has any `zsh` specific features? That script is migrated from Jenkins script and I think zsh was chosen due to a random personal preference. If bash works, maybe we should stick to bash rather than introducing yet another dependency.

pavanbalaji commented:
> > test/mpi/maint/jenkins/set-xfail.sh → test/mpi/maint/jenkins/set-xfail.zsh
> 
> Does this script has any `zsh` specific features? That script is migrated from Jenkins script and I think zsh was chosen due to a random personal preference. If bash works, maybe we should stick to bash rather than introducing yet another dependency.

I didn't add that dependency.  The script was always `zsh`, even though it was named `sh`.  If it's not relying on any `zsh` features, it can be changed to `bash`.  But that's irrelevant to this PR and can be done separately.  This PR corrected the name, just to make it simpler for my script that was setting the copyright headers.

hzhou commented:
In the spirit of reducing boilerplate clutter, do we need the modeline? I use vim and I never had trouble editing without the modeline. In my opinion, we should work on whitespace checker to guard against tab/space mixing or bad indentations -- if those are issues -- rather than dictate editor settings.

hzhou commented:
Similar to renaming shell scripts, I personally likes to name all perl scripts with explicit `.pl` extension.

pavanbalaji commented:
> In the spirit of reducing boilerplate clutter, do we need the modeline? I use vim and I never had trouble editing without the modeline. In my opinion, we should work on whitespace checker to guard against tab/space mixing or bad indentations -- if those are issues -- rather than dictate editor settings.

That's fine, if that's what everyone agrees on.  I can easily add/remove mode lines.  But I want that to be a group decision and not one person's preference.

hzhou commented:
The test failures probably due to the xfail script name change (to `.zsh`).

pavanbalaji commented:
> The test failures probably due to the xfail script name change (to `.zsh`).

This is a very silly design that the jenkins testing infrastructure relies on this file, while the mpich source code itself doesn't.  That file should be moved into the jenkins testing repo.

hzhou commented:
> 
> 
> > The test failures probably due to the xfail script name change (to `.zsh`).
> 
> This is a very silly design that the jenkins testing infrastructure relies on this file, while the mpich source code itself doesn't. That file should be moved into the jenkins testing repo.

I kinda agree. I'll make that change...

hzhou commented:
test:jenkins/ch4/ucx

hzhou commented:
@pavanbalaji I moved the `set-xfail` script back to jenkins-script repo, and it looks like it's working. You may delete the script from mpich.

wesbland commented:
Is this patch backward/forward compatible with the new contributors' agreement or do we all need to sign it first?

pavanbalaji commented:
> Is this patch backward/forward compatible with the new contributors' agreement or do we all need to sign it first?

This change has nothing to do with the new contributors' agreement.  It simply unifies the copyright to one place, removes developer names and adds institute names in the top-level file instead (I did not see credits for individual developers who were not representing their company anywhere).

wesbland commented:
> This change has nothing to do with the new contributors' agreement. It simply unifies the copyright to one place, removes developer names and adds institute names in the top-level file instead (I did not see credits for individual developers who were not representing their company anywhere).

👍 Fine with me. I was just making sure.

pavanbalaji commented:
After some internal discussion, I have removed the mode lines for all files (both emacs and vim).  The intent is that developers will maintain their own formatting requirements within their `.emacs` or `.vimrc` files.

Also removed the `set-xfail.sh` script from the repo.

Rerunning the tests once more:

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

pavanbalaji commented:
Ended up with a conflict with one of the files because of another merge.  Rebased on to `mpich/master`.  Retesting.

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

pavanbalaji commented:
test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

pavanbalaji commented:
@hzhou Can you review this PR?  I don't know what "review" means for such changes that are entirely done through scripts.  I guess you can look at the general intent of the commits?

hzhou commented on COPYRIGHT:
```diff
@@ -6,7 +6,7 @@ which must be included in the prologue of the code and in all source listings
 
 Copyright Notice
- + 2002 University of Chicago
+2020, Argonne National Laboratory
```
We are expected to update this every year, right?

Some projects use a year range, as some of the code are old and some are new. Should we use `2002 - 2020`?

pavanbalaji replied:
We could use a range.  I'll add that.  I need to go back to CVS and find the earliest commit.

hzhou commented on COPYRIGHT:
```diff
@@ -37,3 +37,18 @@ warranty express or implied, or assumes any legal liability or responsibility
+ * Oak Ridge National Laboratory
+ * Sun Microsystems, Lustre group
+ * Dolphin Interconnect Solutions Inc.
+ * Institut Polytechnique de Bordeaux
```
So much better!

pavanbalaji commented on doc/mpich/sm2dot:
```diff
@@ -1,7 +1,9 @@
-#!/usr/bin/env perl
-#
-# sm2dot
```
No, they were an intentional part of my script to clean up useless header content.

wesbland commented on COPYRIGHT:
```diff
@@ -37,3 +37,18 @@ warranty express or implied, or assumes any legal liability or responsibility
+ * Oak Ridge National Laboratory
+ * Sun Microsystems, Lustre group
+ * Dolphin Interconnect Solutions Inc.
+ * Institut Polytechnique de Bordeaux
```
```suggestion
 * Institut Polytechnique de Bordeaux
 
 Contributions copyright their original authors.
```

This is one thing that's missing if we're trying to replicate the per-file headers. I think @pavanbalaji has another version of this text that might be better.

pavanbalaji replied:
I'll add it in shortly.

pavanbalaji commented:
I have marked this PR as do-not-merge.  Whatever changes we make have to be signed off by our legal teams before I push this change in.

pavanbalaji commented:
test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

pavanbalaji commented:
test:mpich/warnings
test:mpich/ch3/most
test:mpich/ch4/most

pavanbalaji commented:
test:mpich/warnings
test:mpich/ch3/most
test:mpich/ch4/most

hzhou commented:
If nothing significant has changed since a month ago, I think I am ready to approve it. I'll just wait for the tests to clear. Meanwhile, @pavanbalaji , if there is any particular area you'd like me to double check, let me know.

pavanbalaji commented:
> If nothing significant has changed since a month ago, I think I am ready to approve it. I'll just wait for the tests to clear. Meanwhile, @pavanbalaji , if there is any particular area you'd like me to double check, let me know.

Nothing particular to review.  Maybe a quick glance is sufficient.  This is almost entirely scripted.  But the tests are not getting triggered for some reason.
